quantum/process_keycode/process_combo.c

```diff
@@ -547,6 +554,16 @@ bool process_combo(uint16_t keycode, keyrecord_t *record) {
 #ifdef COMBO_ONLY_FROM_LAYER
     /* Only check keycodes from one layer. */
     keycode = keymap_key_to_keycode(COMBO_ONLY_FROM_LAYER, record->event.key);
+#else
+
+#    ifdef COMBO_REF_DEFAULT
+    layer          = combo_ref_from_layer(COMBO_REF_DEFAULT);
+#    else
+    layer = combo_ref_from_layer(get_highest_layer(layer_state));
+#    endif
+    if (layer != get_highest_layer(layer_state)) {
+        keycode = keymap_key_to_keycode(layer, record->event.key);
+    }
```

Comment:
```suggestion
    uint8_t highest_layer = get_highest_layer(layer_state);
#    ifdef COMBO_REF_DEFAULT
    layer = combo_ref_from_layer(COMBO_REF_DEFAULT);
#    else
    layer = combo_ref_from_layer(highest_layer);
#    endif
    if (layer != highest_layer) {
        keycode = keymap_key_to_keycode(layer, record->event.key);
    }
```
With this kind of code there is some chance that the compiler would be able to optimize the unused code away if LTO is enabled (without LTO the optimizer won't be able to assume anything about the implementation of a weak function).